lightning/src/chain/channelmonitor.rs

```diff
@@ -1482,11 +1486,11 @@ impl<Signer: Sign> ChannelMonitor<Signer> {
 	}
 
 	/// Returns the set of txids that should be monitored for re-organization out of the chain.
-	pub fn get_relevant_txids(&self) -> Vec<Txid> {
+	pub fn get_relevant_txids(&self) -> Vec<(Txid, BlockHash)> {
 		let inner = self.inner.lock().unwrap();
-		let mut txids: Vec<Txid> = inner.onchain_events_awaiting_threshold_conf
+		let mut txids: Vec<(Txid, BlockHash)> = inner.onchain_events_awaiting_threshold_conf
 			.iter()
-			.map(|entry| entry.txid)
+			.map(|entry| (entry.txid, entry.block_hash.unwrap()))
```

Comment:
Can't this be `None` for older serializations?

Reply:
Hmpf, you're absolutely right, I didn't think about a monitor that just had been restored from an older version. Unfortunately I don't see a good way to retrieve a fallback value, and we can't just skip the entry, so this probably means we have to make the signature `fn get_relevant_txids(&self) -> Vec<(Txid, Option<BlockHash>)>`, which is kind of unfortunate for the interface going forward...